index.js

```diff
@@ -32,6 +37,59 @@ module.exports = {
         ],
       },
     },
+    delete: {
+      validation: {
+        function: 'checkboxValidation',
+        on: 'change',
+      },
+      script: {
+        function: 'fileScript',
+        on: 'change',
+      },
+      type: 'checkbox',
+      label: 'Delete file',
+      options: [
+        { label: 'Delete file',
+          value: 'true',
+        },
+      ],
+      settings: {
+        classes: {
+          hide: 'file--uploaded',
+        },
+      },
+    },
   },
-  html: '<label for="{{file.id}}">{{file.label}}</label><input type="{{file.type}}" id="{{file.id}}" name="{{file.name}}" value="{{file.value}}" {% if settings.types %}{%set comma = joiner() %}accept="{% for type in settings.type %}{{comma()}}{{type}}{% endfor %}{% endif %}" />',
+  html: `<label for="{{file.id}}">{{file.label}}</label>
+    {% set classes = '' %}
+    {% set filepath = '/files' %}
+
+    {% if storage.public %}
+      {% set filepath = storage.public %}
+    {% endif %}
+
+    {% if file.value %}
+      {% set classes = delete.settings.classes.hide %}
```

Comment:
Do we do settings per-input? I thought we only did settings for the overall object